src/test_workflow/integ_test/integ_test_suite.py

```diff
@@ -73,14 +74,20 @@ def __init__(
     def execute_tests(self) -> TestComponentResults:
         pass
 
-    def execute_integtest_sh(self, endpoint: str, port: int, security: bool, test_config: str) -> int:
+    def multi_execute_integtest_sh(self, cluster_endpoints: list, security: bool, test_config: str) -> int:
         script = ScriptFinder.find_integ_test_script(self.component.name, self.repo.working_directory)
         if os.path.exists(script):
-            cmd = f"{script} -b {endpoint} -p {port} -s {str(security).lower()} -v {self.bundle_manifest.build.version}"
+            if len(cluster_endpoints) == 1:
+                single_data_node = cluster_endpoints[0]['data_nodes'][0]
```

Comment:
This should be a list of first class citizens of type `ClusterEndpoint`, and the code would say `next(next(cluster_endpoints).data_nodes)` or `cluster_endpoints[0].data_nodes[0]`.